ignite/metrics/rouge.py

```diff
@@ -0,0 +1,151 @@
+from collections import defaultdict
+from typing import Callable, DefaultDict, Dict, List, Optional, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics import Metric
+
+# These decorators helps with distributed settings
+from ignite.metrics.metric import reinit__is_reduced, sync_all_reduce
+
+
+class Rouge(Metric):
+    r"""Calculates the Rouge Score for two Sequence of Tokens.
+
+
```

Comment:
Cite the reference paper :

Lin, Chin-Yew. 2004. ROUGE: a Package for Automatic Evaluation of Summaries. In Proceedings of the Workshop on Text Summarization Branches Out (WAS 2004), Barcelona, Spain, July 25 - 26, 2004.

http://www.aclweb.org/anthology/W/W04/W04-1013.pdf